internal/store/helper.go

```diff
@@ -0,0 +1,117 @@
+package store
+
+import (
+	"context"
+	"fmt"
+	_ "github.com/golang-migrate/migrate/v4/database/pgx"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/kong/pg-aurora-client/pkg/pool"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	defaultMaxConnections = 50
+	defaultMinConnections = 20
+)
+
+var writeQuery = `UPDATE canary SET id=id +1, ts = CURRENT_TIMESTAMP`
+
+func writeValidator(ctx context.Context, conn *pgxpool.Conn, logger *zap.Logger) bool {
+	exec, err := conn.Exec(ctx, writeQuery)
+	if err != nil {
+		logger.Error("write validation failed", zap.Error(err))
+		return false
+	}
+
+	logger.Info("healthcheck write canary", zap.Int64("rowsUpdated", exec.RowsAffected()))
+	return true
+}
+
+func SetupTestDatabase() (testcontainers.Container, *pool.AuroraPGPool, error) {
+	containerReq := testcontainers.ContainerRequest{
+		Image:        "postgres:latest",
+		ExposedPorts: []string{"5432/tcp"},
+		WaitingFor:   wait.ForListeningPort("5432/tcp"),
+		Env: map[string]string{
```

Comment:
nit: `koko` seems like weird default names for a shared library. I think the default postgres container users `postgres` for everything.

Reply:
Changed